README.md

```diff
@@ -1,5 +1,102 @@
+run ``mvn clean package`` then run ``java -jar target/spring-boot-example-0.0.1-SNAPSHOT.jar server application.yml
+``
+# What is the application?
+REST API monolith Spring application that uses a postgres database to store customers. A REST API is exposed to the user. 
 
-Run ```docker run -p 5432:5432 -e POSTGRES_PASSWORD=password -e POSTGRES_USER=user postgres``` before starting 
-the main Spring Boot application.
+### Endpoints
 
+1. **Get all customers**
+    - **Request:**
+        - Method: GET
+        - URL: `/api/v1/customers`
+        - Example Request:
+          ```
+          GET /api/v1/customers
+          ```
+
+    - **Response:**
+        - Status Code: 200 OK
+        - Body: Array of customer objects containing `id`, `name`, `email`, and `age` fields.
+
+2. **Add a new customer**
+    - **Request:**
+        - Method: POST
+        - URL: `/api/v1/customers`
+        - Example Request:
+          ```
+          POST /api/v1/customers
+          Content-Type: application/json
+   
+          {
+            "name": "John Doe",
+            "email": "john@example.com",
+            "age": 30
+          }
+          ```
+
+    - **Response:**
+        - Status Code: 201 Created
+        - Body: New Customer object containing `id`, `name`, `email`, and `age` fields.
+
+3. **Delete a customer**
+    - **Request:**
+        - Method: DELETE
+        - URL: `/api/v1/customers/{customerId}`
+        - Example Request:
+          ```
+          DELETE /api/v1/customers/123
+          ```
+
+    - **Response:**
+        - Status Code: 200 OK
+        - Body: Success message confirming deletion.
+
+4. **Update a customer**
+    - **Request:**
+        - Method: PUT
+        - URL: `/api/v1/customers/{customerId}`
+        - Example Request:
+          ```
+          PUT /api/v1/customers/123
+          Content-Type: application/json
+   
+          {
+            "name": "Updated Name",
+            "email": "updated@example.com",
+            "age": 35
+          }
+          ```
+
+    - **Response:**
+        - Status Code: 201 Created
+        - Body: New Customer object containing `id`, `name`, `email`, and `age` fields.
+
+ These endpoints are defined in the `CustomerController` class.
+
+### How the Test works
+- The  integration test spins up the controller, service and database layer. As part of the test it uses the TestContainers 
+library to set up a Postgres container.
+- The ``CustomerControllerTest.Initializer`` static class overrides the application properties in ``application.yml`` so that our 
+application points to the Postgres container
+- Next liquibase runs the database migrations defined in the `migrations.xml` file to set up the customer table in the PostgreSQL container so that the Postgres container contains a Customer table.
+- After, JDBI connects to our Postgres container to add test data into our Postgres container (e.g. adding and removing customers from our table) for our tests to work. 
```

Comment:
Reword this to make it a little clearer - e.g. "Our tests use JDBI to connect to the Postgres container to setup test data".